common/file/Util.cpp

```diff
@@ -129,15 +129,16 @@ bool FindMatchingFiles(const string &directory,
   FindClose(h_find);
 #else
   DIR *dp;
-  struct dirent dir_ent;
   struct dirent *dir_ent_p;
   if ((dp = opendir(directory.data())) == NULL) {
     OLA_WARN << "Could not open " << directory << ": " << strerror(errno);
     return false;
   }
 
-  if (readdir_r(dp, &dir_ent, &dir_ent_p)) {
-    OLA_WARN << "readdir_r(" << directory << "): " << strerror(errno);
+  errno = 0;
```

Comment:
Is it worth explicitly checking that errno is non-zero after the opendir call for some weird edge-case install, before we explicitly set it to okay?

Reply:
True. `errno` is thread-local but checking it won't hurt. However, since `opendir` sets `errno` in case of an error (and we are checking the return value for `NULL`), it should not happen

Reply:
You never quite know with some of the weird distros we support.

Reply:
I've investigated a bit more into this. Main points (from https://linux.die.net/man/3/errno):
```
Its value is significant only when the return value of the call indicated an error (i.e., -1 from 
most system calls; -1 or NULL from most library functions); a function that succeeds is
allowed to change errno.

Valid error numbers are all nonzero; errno is never set to zero by any system call or library function.
```

As such, checking it for `0` doesn't make sense since that would only be true if it was `0` from the beginning or was set to `0` intentionally after a last system function call.

However, setting it to `0` is required because the previous, successful function call might have left it with a value.

And checking it for a value != `0` after calling `readdir` is required since the return value of `NULL` can have two meanings:
* Either it was all okay and there are no more entries to be returned OR
* there was an error

In the first case, if errno had not been set to `0` prior to the call, I would not know if it was okay, since (according to the docs), no system function call is allowed to set errno to zero.

Reply:
Given it says:

> a function that succeeds _is_ allowed to change errno.

I'm unclear what a successful opendir call and an errorno value would mean, although it seems to say it shouldn't.

Reply:
Resolving as discussed in main thread now.